### PR TITLE
Add DW mission engine

### PR DESCRIPTION
**Pull request by ImplicitDeny**
Orion is now shipped with a modular mission engine inspired and named after the ancient LM persistent mission Daily Warfare. DW is capable of selecting random scenarios between those marked as playable, and will handle a similar ennemy occupation system as the one found in the original Daily Warfare.

### CODE REVIEW

File: functions/dw_populate/fn_populateSheds.sqf [hunk 1]
```diff
@@ -0,0 +1,125 @@
+﻿//exec: server
+
+params["_center", "_sheds"];
+
+_count = 0;
+
+if(!_sheds) exitWith {_count};
+
+// Demi abri avec mur en béton
```
Comment (ArwynFr) [inline functions/dw_populate/fn_populateSheds.sqf]: Shouldn't you make 3 different files for each type ?

**Comment by ArwynFr** [comment]
Is it relevant to separate task/marker handling, waituntil, etc ... from actual mission code ?
Couldn't that common part be done based on metadata ?
It would prevent buggy missions to interfere

File: functions/dw_populate/fn_populateZone.sqf [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+_miliWatch = ([DW_MISSION_POSITION, DW_MISSION_POPULATE select 0] call DW_fnc_populateMilitaryWatchTowers);
+_sheds = ([DW_MISSION_POSITION, DW_MISSION_POPULATE select 1] call DW_fnc_populateSheds);
+// helis = [DW_MISSION_POSITION, DW_MISSION_POPULATE select 2] call DW_fnc_populateHelipads;
+_solar = ([DW_MISSION_POSITION, DW_MISSION_POPULATE select 3] call DW_fnc_populateSolarTowers);
+_patrols = ([DW_MISSION_POSITION, DW_MISSION_POPULATE select 4] call DW_fnc_populatePatrols);
+
+sleep 7;
+_total = _miliQG + _miliTow + _miliWatch + _sheds + _solar + _patrols;
+
+/*
```
Comment (ArwynFr) [inline functions/dw_populate/fn_populateZone.sqf]: Remove commented code

File: functions/dw_server/fn_markerCreation.sqf [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
```
Comment (ArwynFr) [inline functions/dw_server/fn_markerCreation.sqf]: File header should be nowhere or everywhere :)
Or all public functions etc ...

File: initServer.sqf [hunk 4]
```diff
@@ -0,0 +1,29 @@
+//Init commander
+DW_COMMANDER disableAI "ALL";
+DW_COMMANDER enableAI "ANIM";
+
+/* Blocage des officiers, TODO
```
Comment (ArwynFr) [inline initServer.sqf]: Remove commented code

File: functions/dw_populate/fn_populateSheds.sqf (hunk 1 above)
Comment (ImplicitDeny) [reply functions/dw_populate/fn_populateSheds.sqf]: I should, and I will. However those scripts are being rewritten due to the occupation system feature dev. Once polished I will replace these scripts with the new ones.

File: functions/dw_populate/fn_populateZone.sqf (hunk 2 above)
Comment (ImplicitDeny) [reply functions/dw_populate/fn_populateZone.sqf]: See https://github.com/team-gsri/Orion/pull/15#discussion_r348959136

File: functions/dw_server/fn_markerCreation.sqf (hunk 3 above)
Comment (ImplicitDeny) [reply functions/dw_server/fn_markerCreation.sqf]: That's some old legacy code. This function will probably be properly removed as soon as I'm sure that no mission is using it. Its only role is to replace two or three easy-to-read lines by a single, barely readable, function call. Not quite useful, but was part of the original DW engine, which is now vastly reworked.

File: initServer.sqf (hunk 4 above)
Comment (ImplicitDeny) [reply initServer.sqf]: See https://github.com/team-gsri/Orion/pull/15/commits/5fb60f7d214c619debfde5730e895ee79a3a8f9c

**Comment by ImplicitDeny** [comment]
> Is it relevant to separate task/marker handling, waituntil, etc ... from actual mission code ?
> Couldn't that common part be done based on metadata ?
> It would prevent buggy missions to interfere

I don't understand this question. The PR will bring a mission system designed to manage tasks and zone marker based on metadata. It could also have managed the win condition, however some specific complex missions can take advantage of managing this directly, so it was left to the mission designer's care.

What do you suggest ? Moving the win/lose condition out of the mission perimeter, and placing it directly in the DW engine ?

File: functions/dw_populate/fn_populateSheds.sqf (hunk 1 above)
Comment (ArwynFr) [reply functions/dw_populate/fn_populateSheds.sqf]: Resolution through #16

File: functions/dw_populate/fn_populateZone.sqf (hunk 2 above)
Comment (ArwynFr) [reply functions/dw_populate/fn_populateZone.sqf]: Resolution through #17

Comment (ArwynFr) [reply functions/dw_populate/fn_populateZone.sqf]: Resolution through #18

File: functions/dw_server/fn_markerCreation.sqf (hunk 3 above)
Comment (ArwynFr) [reply functions/dw_server/fn_markerCreation.sqf]: Resolution through #18

**Comment by ArwynFr** [comment]
It's not much about the win/loss condition but rather about the marker / journal task handling.
I think this feature should not be in the mission scope but rather be handled by the DW engine.

**Comment by ImplicitDeny** [comment]
> not be in the mission scope but rather be handled by the DW engine.
Well... It is. O_o
See https://github.com/team-gsri/Orion/blob/5fb60f7d214c619debfde5730e895ee79a3a8f9c/functions/dw_engine/fn_launchMainMission.sqf#L19 and following lines. It's all about extracting selected mission's metadata in order to create the appropriate task and map marker.

**Comment by ArwynFr** [comment]
My point is that lines such as https://github.com/team-gsri/Orion/blob/5fb60f7d214c619debfde5730e895ee79a3a8f9c/fob/ile/main/dummy/dummy1.sqf#L3 shall not be in the mission file.
createMarker and BIS_fnc_taskSetType should be called by the engine, not the mission.
It would limit copy&paste&fail code wrecking markers/tasks if something is coded wrong.

**Comment by ImplicitDeny** [comment]
Heck, this is by no means a valid file ! It's just the old file structure, legacy from LM's DW, which I used in the very first hours of living of the dw branch, but this has to be removed. I just forget to delete this folder, gonna do it RN.

**Pull request merged**